### ISSUE TITLE
Can't update from HACS

### ISSUE THREAD

**Issue by bschatzow** [bruxy70/Garbage-Collection#50]
Says update available, I hit update, looks like it was installed. Says update available after installing.

**Comment by bruxy70** [comment]
What is the installed version it says? The latest is 2.7. Does it remain that even after the restart? 
(it should ask for restart in the header

**Comment by bschatzow** [comment]
It says I am on 2.6.  It doesn't ask for a restart.  I update and restart and let you know.

**Comment by bschatzow** [comment]
Did a restart and still on 2.6

**Comment by bruxy70** [comment]
I am sorry, I cannot replicate the issue. Works fine here. Do you have the last version of HACS? Does it show any error? Have you tried the reinstall button?

**Comment by bschatzow** [comment]
I have hacs .174, I have tried the reinstall and no difference.  Maybe uninstall and then reinstall?  Saw this on the last update and did a manual download and copy.  Maybe this caused the issue?

**Comment by bruxy70** [comment]
I guess that could be it. Please try to uninstall, remove it and install from HACS. Let me know. Thanks.

**Comment by Woodcat64** [comment]
I'm having same issues. It started when v2.1 came out.  Every new release was showing "update available". When I pressed update, it looked like it's updating, showed me message to restart HA, but the version still showed v2.1, even after restart.
When v2.5 came, I decided to uninstall Garbage-Collection from HACS, restart and reinstalled and it worked. Now i'm on v2.5, using integration install method.
I was not able update to 2.6 or 2.7 again. Even if I uninstall and install v2.7, v2.5 gets installed.
HACS v0.17.4
Beside reinstalling HACS, I'm not sure what else to do. 
![GC](https://user-images.githubusercontent.com/50119854/70095482-dea75a80-15f2-11ea-9965-cd2356873b6d.PNG)
[Garbage Collection.txt](https://github.com/bruxy70/Garbage-Collection/files/3919020/Garbage.Collection.txt)

**Comment by bruxy70** [comment]
Ok, I am asking HACS to help. Is there anything in the log file?
Did you try to uninstall garbage_collection, make sure it is removed from the custom_components directory, restart HA and install it again from HACS?

**Comment by Woodcat64** [comment]
I just tried this now. Uninstalled garbage_collection, made sure it is removed from the custom_components directory, restarted HA, (showed v2.7 available) and install it again from HACS. When install is done, I get message "Restart pending", but it already shows "installed v2.5,  v2.7 available". Restart anyway, still  shows "installed v2.5"
Here is the home-assistant.log
[Garbage Collection1.txt](https://github.com/bruxy70/Garbage-Collection/files/3924618/Garbage.Collection1.txt)

**Comment by bruxy70** [comment]
Ok, I have checked with ludeeus, he thinks it could be a network or storage issue.
You can check it out here: https://github.com/hacs/integration/issues/839#issuecomment-562259303
I was looking in the .storage folder, found two files where HACS keeps the list of repositories. The list of repositories showed the correct version. The installed file shows the version when it was installed (2.2 in my case) and it does not update, which seems to be ok. So I guess you can check if you have the same.
I am afraid that's all I can do.

**Comment by bschatzow** [comment]
Not sure from what you say above how to fix this.  I have plenty of space available.  Two other HAC integrations updated correctly today.

**Comment by bruxy70** [comment]
I just quoted what @ludeeus said. I have no idea how HACS deals with the versions I am affraid.

**Comment by bschatzow** [comment]
2.8 just installed.  Now showing a drop down menu asking what version to install.

**Comment by Woodcat64** [comment]
I got it working. Like bschatzow, I have no storage or network issues. But that led me to .storage and digging through hacs.repositories I noticed in the "bruxy70/Garbage-Collection" section there is a value "selected_tag": v2.5. All of the other integrations have there "selected_tag": null.
By changing that, it broke the Garbage-Collection addon completely as it could not connect to servers  ( it was 404 on this link https://api.github.com/repos/bruxy70/Garbage-Collection/contents/custom_components ). I manually removed the Garbage-Collection from custom_components (since uninstall would not work), changed "installed_commit": to null as well, restarted HA. Reinstall the addon. 
All works now, I'm on v2.7. My guess, the repo file got  misconfigured somehow.
